javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,54 @@
+var $h1 = document.querySelector('h1');
+var $h3 = document.querySelector('h3');
+var incorrectCounter = 0;
+var word = 'grumpy wizards make toxic brew';
+for (var x = 0; x < word.length; x++) {
+  $h1.appendChild(makeSpan(word[x]));
+}
+
+var $span = document.querySelectorAll('span');
+document.addEventListener('keydown', checkKey);
+var $playAgain = document.querySelector('#play-again');
+
+startGame();
+var $button = document.querySelector('button');
+$button.addEventListener('click', startGame);
+
+function startGame() {
+  $span[0].className = 'current';
+  $playAgain.className = 'row center hide';
+  incorrectCounter = 0;
+  for (var z = 1; z < word.length; z++) {
+    $span[z].className = '';
+  }
+}
+
+function makeSpan(string) {
+  var createSpan = document.createElement('span');
+  createSpan.textContent = string;
+  return createSpan;
+}
+
+function checkKey(event) {
+  for (var y = 0; y < $span.length; y++) {
+    if ($span[y].matches('.current') || $span[y].matches('.wrong')) {
+      if (event.key === $span[y].innerText) {
+        $span[y].className = 'correct';
+        if ($span[word.length - 1].className === 'correct') {
+          endGame();
+        }
+        $span[y + 1].className = 'current';
+        return;
+      } else if (event.key !== 'Shift' && event.key !== 'CapsLock') {
+        $span[y].className = 'wrong';
+        incorrectCounter++;
+        return;
+      }
+    }
+  }
```

Comment:
Instead of looping through the entire span list, try declaring a global `currentIndex` variable and only increment it if the user gets the current key right.

You shouldn't need a loop for this challenge except maybe when you add the span elements to the page in the beginning.